File: androidweekview/src/main/java/com/alamkanak/weekview/SplitRect.java
package com.alamkanak.weekview;

import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.RectF;

class SplitRect extends RectF {
	private float division;

	SplitRect(RectF baseRect, float division) {
		super(baseRect);
		this.division = division;
	}

	void drawTo(Canvas canvas, float rx, float ry, Paint topPaint, Paint bottomPaint) {
		if (division == 0) {
			canvas.drawRoundRect(this, rx, ry, bottomPaint);
		} else if (division == 1) {
			canvas.drawRoundRect(this, rx, ry, topPaint);
		} else {
			canvas.drawRoundRect(this, rx, ry, topPaint);
			canvas.save();
			canvas.clipRect(left, top + (bottom - top) * division, right, bottom);
			canvas.drawRoundRect(this, rx, ry, bottomPaint);
			canvas.restore();
		}
	}
}

File: weekview/src/main/java/com/sapuseven/untis/views/weekview/SplitRect.kt
package com.sapuseven.untis.views.weekview

import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.RectF

class SplitRect(baseRect: RectF, private val division: Float) : RectF(baseRect) {
	fun drawTo(canvas: Canvas, rx: Float, ry: Float, topPaint: Paint, bottomPaint: Paint) {
		with(canvas) {
			when (division) {
				0f -> drawRoundRect(this@SplitRect, rx, ry, bottomPaint)
				1f -> drawRoundRect(this@SplitRect, rx, ry, topPaint)
				else -> {
					drawRoundRect(this@SplitRect, rx, ry, topPaint)
					save()
					clipRect(left, top + (bottom - top) * division, right, bottom)
					drawRoundRect(this@SplitRect, rx, ry, bottomPaint)
					restore()
				}
			}
		}
	}
}
